addon/-private/system/store.js

```diff
@@ -547,16 +565,21 @@ Store = Service.extend({
       return this.scheduleFetch(internalModel, options);
     }
 
-    // Refetch the record if the adapter thinks the record is stale
+    if (options.backgroundReload === false) {
+      return Promise.resolve(internalModel);
+    }
```

Comment:
I tend to think that this needs to be moved below the code which checks if a reload should be done (either via the options or via the adapters' `shouldReloadRecord`). In my understanding a reload should take precedence over a background reload.

IMHO the logic should be:

- if we should do a reload (either configured via the passed option or via the adapters' `shouldReloadRecord`), then return a promise which resolves once the record has been reloaded
- if we shouldn't do a reload but we should do a background reload (either configured via the passed option or via , we return a promise which resolves immediately but the record is reloaded in the background.

What do you think?